common/djangoapps/util/milestones_helpers.py

```diff
@@ -212,7 +212,8 @@ def get_required_content(course_key, user):
     and if those milestones can be fulfilled via completion of a particular course content module
     """
     required_content = []
-    if settings.FEATURES.get('MILESTONES_APP'):
+    # TODO This will probably pose issues for courses that rely on milestones to gate content.
+    if settings.FEATURES.get('MILESTONES_APP') and user.is_authenticated():
```

Comment:
Possibly. I am still digging into the potential implications of this.

Reply:
I have opted to raise an error when attempting to get milestones with an anonymous user. This may potentially be a poor UX for anonymous viewers who show up, but it will ensure that we don't exposed the locked content. Additionally, this problem will only manifest if someone shares a link and later adds a milestone.